vsintegration/src/FSharp.ProjectSystem.Base/Project/FSharpEditorFactory.cs

```diff
@@ -16,12 +16,15 @@
 
 namespace Microsoft.VisualStudio.FSharp.ProjectSystem
 {
+    // 64 represents a hex number. It needs to be greater than 37 so the TextMate editor will not be chosen as higher priority.
     [Guid(Constants.FSharpEditorFactoryIdString)]
     [ProvideEditorFactory(typeof(FSharpEditorFactory), 101, CommonPhysicalViewAttributes = (int)__VSPHYSICALVIEWATTRIBUTES.PVA_SupportsPreview)]
-    [ProvideEditorExtension(typeof(FSharpEditorFactory), ".fs", 32)]
-    [ProvideEditorExtension(typeof(FSharpEditorFactory), ".fsi", 32)]
-    [ProvideEditorExtension(typeof(FSharpEditorFactory), ".fsscript", 32)]
-    [ProvideEditorExtension(typeof(FSharpEditorFactory), ".fsx", 32)]
+    [ProvideEditorExtension(typeof(FSharpEditorFactory), ".fs", 64)]
+    [ProvideEditorExtension(typeof(FSharpEditorFactory), ".fsi", 64)]
+    [ProvideEditorExtension(typeof(FSharpEditorFactory), ".fsscript", 64)]
+    [ProvideEditorExtension(typeof(FSharpEditorFactory), ".fsx", 64)]
+    [ProvideEditorExtension(typeof(FSharpEditorFactory), ".ml", 64)]
+    [ProvideEditorExtension(typeof(FSharpEditorFactory), ".mli", 64)]
```

Comment:
Can/should we file a bug on the VSSDK to warn if there are multiple `ProvideEditorExtension` attributes with different priorities?

Reply:
I am not sure. I imagine in other projects their may be different priorities of editors.

Reply:
The same guid and the same extension multiple times with different priorities, yes, that would be something to warn.

Reply:
This already happens in ActivityLog - it warns when it sees the same registry key set with different values, however, this isn't the case here - we had two different editor factories.